LeetCodeSolutions/ReorderRoutesToLeadToZero/ReorderRoutesToLeadToZero.cs

```diff
@@ -0,0 +1,59 @@
+namespace LeetCodeSolutions;
+public class ReorderRoutesToLeadToZeroSolution
+{
+    private class Connection(int neighbor, bool correctionNeeded)
+    {
+        public int Neighbor = neighbor;
+        public bool CorrectionNeeded = correctionNeeded;
+    }
+
+    private class GraphSearcher
+    {
+        public int Count;
+
+        private readonly List<List<Connection>> adjacents;
+        public GraphSearcher(int n, ref int[][] connections)
+        {
+            Count = 0;
+            adjacents = [];
+
+            for (int i = 0; i < n; i++) { adjacents.Add([]); }
+
+            foreach (int[] connection in connections)
+            {
+                AddConnection(0, connection);
```

Comment:
make some self documenting functions, like "add from- node connection" and "add to-node connection".